### PR TITLE
Update the documentation around headers.

### PR DESCRIPTION
**Pull request by BrynCooke**
The headers layer is now a headers plugin. Layers are gone in the configuration.
Update the docs to match.

### CODE REVIEW

**Comment by bnjjj** [comment]
Regarding this PR https://github.com/apollographql/router/pull/600 could you directly use the name `apollo.headers` in the docs please ?

**Comment by BrynCooke** [comment]
> Regarding this PR #600 could you directly use the name `apollo.headers` in the docs please ?

Done, nice catch!

**Review by bnjjj — APPROVED** [review]

File: docs/source/configuration.mdx [hunk 1]
```diff
@@ -78,6 +78,23 @@ Default: `info`
...
+router --schema > configuration_schema.json
+```
+
+One you have generated the schema configure your text editor. Here are some instructions for well known editors:
+* [Emacs](https://emacs-lsp.github.io/lsp-mode/page/lsp-yaml)
+* [IntelliJ](https://www.jetbrains.com/help/idea/json.html#ws_json_using_schemas)
+* [Sublime](https://github.com/sublimelsp/LSP-yaml)
+* [Vim](https://github.com/Quramy/vison)
+* [VsCode](https://code.visualstudio.com/docs/languages/json#_json-schemas-and-settings)
+
```
Comment (Geal) [inline docs/source/configuration.mdx:97]: awesome, this whole section will be really helpful!

File: docs/source/configuration.mdx [hunk 2]
```diff
@@ -312,21 +245,23 @@ Enables you to add custom headers to requests going to a specific subgraph. Thes
...
+    - remove:
+        name: "x-legacy-account-id"
+    # Subgraph specific configuration
+    subgraphs:
+      products:
+        # Calls to the products subgraph will have the "router-subgraph-name" header.
+        - insert:
+            name: "router-subgraph-name"
+            value: "products"
 ```
```
Comment (Geal) [inline docs/source/configuration.mdx:272]: could you add an example with anchors, to show how to avoid repetition?

Comment (Geal) [reply docs/source/configuration.mdx:272]: great, thanks!

**Review by Geal — APPROVED** [review]

**Pull request merged**